### PR TITLE
1332 workflow restrictions

### LINKED ISSUE #1332: Support workflow restrictions in `github_actions_runner_group`
Support the following options in `resource github_actions_runner_group`

```
restricted_to_workflows = boolean
selected_workflows = Array<string>
```

### Affected Resource(s)
- github_actions_runner_group

### References
- Github blog on Workflow Access for self-hosted runners: https://github.blog/changelog/2022-03-21-github-actions-restrict-self-hosted-runner-groups-to-specific-workflows/
- Github API Documentation: https://docs.github.com/en/rest/actions/self-hosted-runner-groups#get-a-self-hosted-runner-group-for-an-organization
- Terraform resource in question: https://registry.terraform.io/providers/integrations/github/latest/docs/resources/actions_runner_group

### PR DESCRIPTION
**Pull request by bob-bins**
Resolves #1332 

----

## Behavior

Supports the following options in resource `github_actions_runner_group`

```
restricted_to_workflows bool
selected_workflows []string
```

----

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:  

* N/A

### Pull request type

 
 

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

File: github/resource_github_actions_runner_group_test.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ func TestAccGithubActionsRunnerGroup(t *testing.T) {
 			resource "github_actions_runner_group" "test" {
 			  name       = github_repository.test.name
 			  visibility = "all"
+				restricted_to_workflows = true
```
Comment (bob-bins) [inline github/resource_github_actions_runner_group_test.go:29]: i don't think i can run these tests... i think they require an enterprise account?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -218,7 +218,7 @@ Once the token has been created, it must be exported in your environment as `GIT
 
 If you do not have an organization already that you are comfortable running tests against, you will need to [create one](https://help.github.com/en/articles/creating-a-new-organization-from-scratch). The free "Team for Open Source" org type is fine for these tests. The name of the organization must then be exported in your environment as `GITHUB_ORGANIZATION`.
 
-Make sure that your organization has a `terraform-module-template` repository ([terraformtesting/terraform-template-module](https://github.com/terraformtesting/terraform-template-module) is an example you can clone) and that its "Template repository" item in Settings is checked.
+Make sure that your organization has a `terraform-template-module` repository ([terraformtesting/terraform-template-module](https://github.com/terraformtesting/terraform-template-module) is an example you can clone) and that its "Template repository" item in Settings is checked.
```
Comment (kfcampbell) [inline CONTRIBUTING.md:221]: Oh gosh, nice catches!

File: github/resource_github_actions_runner_group_test.go (hunk 1 above)
Comment (kfcampbell) [reply github/resource_github_actions_runner_group_test.go:29]: On main, I'm getting a failure on these acceptance tests:

```
--- FAIL: TestAccGithubActionsRunnerGroup (47.67s)
    --- FAIL: TestAccGithubActionsRunnerGroup/creates_runner_groups_without_error (12.52s)
        --- SKIP: TestAccGithubActionsRunnerGroup/creates_runner_groups_without_error/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubActionsRunnerGroup/creates_runner_groups_without_error/with_an_individual_account (0.00s)
        --- FAIL: TestAccGithubActionsRunnerGroup/creates_runner_groups_without_error/with_an_organization_account (12.52s)
    --- FAIL: TestAccGithubActionsRunnerGroup/manages_runner_visibility (11.71s)
        --- SKIP: TestAccGithubActionsRunnerGroup/manages_runner_visibility/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubActionsRunnerGroup/manages_runner_visibility/with_an_individual_account (0.00s)
        --- FAIL: TestAccGithubActionsRunnerGroup/manages_runner_visibility/with_an_organization_account (11.71s)
    --- FAIL: TestAccGithubActionsRunnerGroup/imports_an_all_runner_group_without_error (11.47s)
        --- SKIP: TestAccGithubActionsRunnerGroup/imports_an_all_runner_group_without_error/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubActionsRunnerGroup/imports_an_all_runner_group_without_error/with_an_individual_account (0.00s)
        --- FAIL: TestAccGithubActionsRunnerGroup/imports_an_all_runner_group_without_error/with_an_organization_account (11.47s)
    --- PASS: TestAccGithubActionsRunnerGroup/imports_a_private_runner_group_without_error (0.00s)
        --- SKIP: TestAccGithubActionsRunnerGroup/imports_a_private_runner_group_without_error/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubActionsRunnerGroup/imports_a_private_runner_group_without_error/with_an_individual_account (0.00s)
        --- SKIP: TestAccGithubActionsRunnerGroup/imports_a_private_runner_group_without_error/with_an_organization_account (0.00s)
    --- FAIL: TestAccGithubActionsRunnerGroup/imports_a_selected_runner_group_without_error (11.98s)
        --- SKIP: TestAccGithubActionsRunnerGroup/imports_a_selected_runner_group_without_error/with_an_individual_account (0.00s)
        --- FAIL: TestAccGithubActionsRunnerGroup/imports_a_selected_runner_group_without_error/with_an_organization_account (11.98s)
FAIL
```

On this branch, I'm seeing a panic on line 179 of github/resource_github_actions_runner_group.go:

`d.Set("selected_workflows", runnerGroup.SelectedWorkflows)`. 

The error reads `panic: runtime error: invalid memory address or nil pointer dereference`. Can you reproduce that when you run these tests on the branch locally?

File: github/resource_github_actions_runner_group.go [hunk 3]
```diff
@@ -118,6 +118,8 @@ func resourceGithubActionsRunnerGroupCreate(d *schema.ResourceData, meta interfa
 	d.Set("selected_repositories_url", runnerGroup.GetSelectedRepositoriesURL())
 	d.Set("visibility", runnerGroup.GetVisibility())
 	d.Set("selected_repository_ids", selectedRepositoryIDs) // Note: runnerGroup has no method to get selected repository IDs
+	d.Set("restricted_to_workflows", runnerGroup.GetRestrictedToWorkflows())
```
Comment (kfcampbell) [inline github/resource_github_actions_runner_group.go:121]: These additional properties should be identified in the appropriate docs page for this resource.

File: github/resource_github_actions_runner_group_test.go (hunk 1 above)
Comment (bob-bins) [reply github/resource_github_actions_runner_group_test.go:29]: I'm not able to successfully run the `TestAccGithubActionsRunnerGroup` tests even from the `main` branch, so it makes it difficult for me to verify my own PR (other tests work file however). Hm...

Comment (kfcampbell) [reply github/resource_github_actions_runner_group_test.go:29]: Ohh that makes sense. We currently don't have a framework in place for enterprise testing, although it's on the TODO list in the effort starting with #1414.

File: website/docs/r/actions_runner_group.html.markdown [hunk 4]
```diff
@@ -42,6 +42,8 @@ The following arguments are supported:
 * `selected_repository_ids`    - List of repository IDs that can access the runner group
 * `selected_repositories_url`  - Github API URL for the runner group's repositories
 * `visibility`                 - The visibility of the runner group
+* `restricted_to_workflows`    - If true, the runner group will be restricted to running only the workflows specified in the selected_workflows array
```
Comment (kfcampbell) [inline website/docs/r/actions_runner_group.html.markdown:45]: Is the default false here? Would you mind noting that?

**Review by nickfloyd — APPROVED** [review]
Hey @bob-bins thanks again for the contributions here! ❤️

**Pull request merged**